### DISCUSSION TITLE
Allow manual checkpointing

### DISCUSSION THREAD

**Discussion by northeastprince** [sparklemotion/sqlite3-ruby#681] [Ideas]
It'd be nice if there was a way to [manually checkpoint the WAL file](https://www.sqlite.org/wal.html#:~:text=Application%2DInitiated%20Checkpoints) if you need everything in one file at a certain moment.

**Comment by flavorjones** [comment]
@northeastprince Thanks for asking about this. Can you share a bit more information about your use case? Adding a wrapper for `sqlite3_wal_checkpoint_v2()` wouldn't be hard, but I do want to know what it's being used before committing to adding it to the gem.

**Comment by flavorjones** [comment]
You may be able to use the pragma for this: https://www.sqlite.org/pragma.html#pragma_wal_checkpoint
